networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/NodeIdToDataColumnSidecarSubnetsCalculator.java

```diff
@@ -47,7 +47,8 @@ static NodeIdToDataColumnSidecarSubnetsCalculator create(
                     final List<UInt64> subnets =
                         MiscHelpersFulu.required(version.miscHelpers())
                             .computeDataColumnSidecarBackboneSubnets(
-                                nodeId, groupCount.orElse(config.getCustodyRequirement()));
+                                peerId.toUInt256(),
```

Comment:
it should be based on discovery nodeId, I see it will return something from libp2p nodeId for existing nodes which is incorrect, you should get discoveryNodeId from existing peer here: https://github.com/Consensys/teku/blob/f69d337c818c902fb13d10e540fccbac17acafaf/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/DefaultEth2Peer.java#L197-L200